firestore.rules

```diff
@@ -2036,6 +2042,60 @@ service cloud.firestore {
       allow delete: if checkDeleteQueueEntity(request.auth);
     }
 
+    // Roles implementation
+    // -----------------------------
+
+    function getRoleAfter(id, index) {
+      return getAfter(/$(databasePath())/UserRoles/$(id)/Roles/$(index))
+    }
+
+    function getRoleBefore(id, index) {
+      return get(/$(databasePath())/UserRoles/$(id)/Roles/$(index))
+    }
+
+    function roleExistsAfter(id, index) {
+      return existsAfter(/$(databasePath())/UserRoles/$(id)/Roles/$(index))
+    }
+
+    function roleExistsBefore(id, index) {
+      return exists(/$(databasePath())/UserRoles/$(id)/Roles/$(index))
+    }
+
+    function validateRole(resource) {
+      return isString(resource.data.roleName) &&
+      resource.id == resource.data.roleName &&
+      userExistsBefore(resource.data.userId) &&
+      (resource.data.roleName == 'admin' || resource.data.roleName == 'hil')
+    }
+
+    function checkReadRole(auth, resource) {
+      return currentUserCanReadPrivateEntity(auth, resource, resource.userId)
+    }
+
+    function checkCreateRole(auth, id, index) {
+      let role = getRoleAfter(id, index);
+      return role != null && validateRole(role) &&
+        currentUserIsServiceAccount(auth)
+    }
+
+    function checkUpdateRole(auth, id, index) {
+      let queueEntity = getRoleAfter(id, index);
```

Comment:
think this was meant to be `let role = ...`

Reply:
of course 😅